From 7f489bea22590da7ff1fb39adad23527ee7bed9b Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Thu, 5 Dec 2013 10:54:10 -0800
Subject: [PATCH 098/122] MIPS: oct_ilm: Fix debugfs file permissions.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: 2808f39ae6a1ca8712eef542a79e465a3d0ed141
Description:

Read-only files shouldn't have write-only permissions.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 arch/mips/cavium-octeon/oct_ilm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/mips/cavium-octeon/oct_ilm.c b/arch/mips/cavium-octeon/oct_ilm.c
index 71b213d..3afe52d 100644
--- a/arch/mips/cavium-octeon/oct_ilm.c
+++ b/arch/mips/cavium-octeon/oct_ilm.c
@@ -72,14 +72,14 @@ static int init_debufs(void)
 		return -1;
 	}
 
-	show_dentry = debugfs_create_file("statistics", 0222, dir, NULL,
+	show_dentry = debugfs_create_file("statistics", S_IRUGO, dir, NULL,
 					  &oct_ilm_ops);
 	if (!show_dentry) {
 		pr_err("oct_ilm: failed to create debugfs entry oct_ilm/statistics\n");
 		return -1;
 	}
 
-	show_dentry = debugfs_create_file("reset", 0222, dir, NULL,
+	show_dentry = debugfs_create_file("reset", S_IWUSR | S_IWGRP, dir, NULL,
 					  &reset_statistics_ops);
 	if (!show_dentry) {
 		pr_err("oct_ilm: failed to create debugfs entry oct_ilm/reset\n");
-- 
1.8.2.1

